tools/ktx/basis_utils.h

```diff
@@ -203,72 +167,56 @@ struct OptionsCodec {
-        ClampedOption<ktx_uint32_t> qualityLevel;
-        ClampedOption<ktx_uint32_t> maxEndpoints;
-        ClampedOption<ktx_uint32_t> maxSelectors;
-        ClampedOption<ktx_uint32_t> uastcRDODictSize;
-        ClampedOption<float> uastcRDOQualityScalar;
-        ClampedOption<float> uastcRDOMaxSmoothBlockErrorScale;
-        ClampedOption<float> uastcRDOMaxSmoothBlockStdDev;
-
-        BasisOptions() :
-            threadCount(ktxBasisParams::threadCount, 1, 10000),
-            qualityLevel(ktxBasisParams::qualityLevel, 1, 255),
-            maxEndpoints(ktxBasisParams::maxEndpoints, 1, 16128),
-            maxSelectors(ktxBasisParams::maxSelectors, 1, 16128),
-            uastcRDODictSize(ktxBasisParams::uastcRDODictSize, 256, 65536),
-            uastcRDOQualityScalar(ktxBasisParams::uastcRDOQualityScalar,
-                                    0.001f, 50.0f),
-            uastcRDOMaxSmoothBlockErrorScale(
-                            ktxBasisParams::uastcRDOMaxSmoothBlockErrorScale,
-                            1.0f, 300.0f),
-            uastcRDOMaxSmoothBlockStdDev(
-                            ktxBasisParams::uastcRDOMaxSmoothBlockStdDev,
-                            0.01f, 65536.0f)
-        {
-            uint32_t tc = std::thread::hardware_concurrency();
-            if (tc == 0) tc = 1;
-            threadCount.max = tc;
-            threadCount = tc;
-
-            structSize = sizeof(ktxBasisParams);
-            // - 1 is to match what basisu_tool does (since 1.13).
-            compressionLevel = KTX_ETC1S_DEFAULT_COMPRESSION_LEVEL - 1;
-            qualityLevel.clear();
-            maxEndpoints.clear();
-            endpointRDOThreshold = 0.0f;
-            maxSelectors.clear();
-            selectorRDOThreshold = 0.0f;
-            normalMap = false;
-            separateRGToRGB_A = false;
-            preSwizzle = false;
-            noEndpointRDO = false;
-            noSelectorRDO = false;
-            uastc = false; // Default to ETC1S.
-            uastcRDO = false;
-            uastcFlags = KTX_PACK_UASTC_LEVEL_DEFAULT;
-            uastcRDODictSize.clear();
-            uastcRDOQualityScalar.clear();
-            uastcRDODontFavorSimplerModes = false;
-            uastcRDONoMultithreading = false;
-            noSSE = false;
-            verbose = false; // Default to quiet operation.
-            for (int i = 0; i < 4; i++) inputSwizzle[i] = 0;
-        }
-    };
+
+    // The remaining numeric fields are clamped within the Basis library
+    ClampedOption<ktx_uint32_t> qualityLevel;
+    ClampedOption<ktx_uint32_t> maxEndpoints;
+    ClampedOption<ktx_uint32_t> maxSelectors;
+    ClampedOption<ktx_uint32_t> uastcRDODictSize;
+    ClampedOption<float> uastcRDOQualityScalar;
+    ClampedOption<float> uastcRDOMaxSmoothBlockErrorScale;
+    ClampedOption<float> uastcRDOMaxSmoothBlockStdDev;
+
+    OptionsBasis() :
+        qualityLevel(ktxBasisParams::qualityLevel, 1, 255),
```

Comment:
Don't we still need to initialize `threadCount` and `noSSE` here? They are part of ktxBasisParams and this is the constructor.

Reply:
Fixed in https://github.com/KhronosGroup/KTX-Software/pull/817/commits/dd12a5ec82f53758c77cbf18c7c2f6e2443a30ff